Automattic/jetpack
# Connection: Soft disconnect cleanup/refactor of deprecated method calls

#### Changes proposed in this Pull Request:
Implement the changes outlined for the last three tasks in #24654 by removing calls to a number of deprecated methods in Connection. The changes will affect the following:
* Backup
* Jetpack

#### Other information:


#### Jetpack product discussion
* p1659019437010529-slack-C02NENKA4GN

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
##### Connected Plugins
1. Deactivate all Jetpack plugins other than Jetpack.
2. Go to the "Jetpack Debug" plugin, activate "REST API Tester", and send a `GET` request to `connection/plugins`.
3. Confirm that the output looks as follows:
```
[
    {
        "name": "Jetpack",
        "slug": "jetpack"
    }
]
```
4. Activate the "Jetpack Boost" plugin.
5. Rerun the "REST API Tester" test by sending a `GET` to `connection/plugins`.
6. Confirm that the output looks as follows:
```
[
    {
        "name": "Jetpack",
        "slug": "jetpack"
    },
    {
        "name": "Jetpack Boost",
        "url_info": "",
        "slug": "jetpack-boost"
    }
]
```

##### Test Only Connected Plugin Check
1. Deactivate all Jetpack plugins other than Jetpack and Jetpack Boost.
2. Go to Jetpack > My Jetpack and confirm that a full connection is established (both site and user). If it is not, fully connect Jetpack.
3. Deactivate the Jetpack plugin.
4. Go to Jetpack > My Jetpack and confirm that Jetpack is fully connected.
5. Activate the Jetpack plugin.
6. Deactivate the Jetpack Boost plugin.
7. Verify that the Jetpack plugin is active and all other Jetpack plugins are deactivated.
8. Go to Jetpack > My Jetpack and confirm that Jetpack is fully connected.
9. Deactivate the Jetpack plugin.
10. Activate the Jetpack plugin.
11. You should now see a page asking you to set up Jetpack.
